config/routes/admin.rb

```diff
@@ -71,7 +71,7 @@
 
     constraints FeatureToggleConstraint.new(:admin_style_v3) do
       resources :products, to: 'products_v3#index', only: :index do
-        patch :bulk_update, on: :collection
+        patch :bulk_update, to: 'products_v3#bulk_update', as: :bulk_update
```

Comment:
Maybe this, I think you forgot the `collection`
```suggestion
       collection do
         patch :bulk_update, to: 'products_v3#bulk_update', as: :bulk_update
       end
```